include/alpaka/exec/ExecCpuOmp2Threads.hpp

```diff
@@ -169,27 +169,25 @@ namespace alpaka
                         // Therefore we use 'omp parallel' with the specified number of threads in a block.
                         #pragma omp parallel num_threads(iBlockThreadCount)
                         {
-#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
-                            // GCC 5.1 fails with:
-                            // error: redeclaration of const int& iBlockThreadCount
-                            // if(numThreads != iBlockThreadCount
-                            //                ^
-                            // note: const int& iBlockThreadCount previously declared here
-                            // #pragma omp parallel num_threads(iBlockThreadCount)
-                            //         ^
-#if (!BOOST_COMP_GNUC) || (BOOST_COMP_GNUC < BOOST_VERSION_NUMBER(5, 0, 0))
                             // The first thread does some checks in the first block executed.
                             if((::omp_get_thread_num() == 0) && (acc.m_gridBlockIdx.sum() == 0u))
                             {
                                 int const numThreads(::omp_get_num_threads());
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
                                 std::cout << BOOST_CURRENT_FUNCTION << " omp_get_num_threads: " << numThreads << std::endl;
-                                if(numThreads != iBlockThreadCount)
-                                {
-                                    throw std::runtime_error("The OpenMP 2.0 runtime did not use the number of threads that had been required!");
-                                }
-                            }
 #endif
+                                // GCC 5.1 fails with:
+                                // error: redeclaration of const int& iBlockThreadCount
+                                // (numThreads == iBlockThreadCount)
+                                //                ^
+                                // note: const int& iBlockThreadCount previously declared here
+                                // #pragma omp parallel num_threads(iBlockThreadCount)
+                                //         ^
+#if (!BOOST_COMP_GNUC) || (BOOST_COMP_GNUC < BOOST_VERSION_NUMBER(5, 0, 0)) || (BOOST_COMP_GNUC >= BOOST_VERSION_NUMBER(6, 0, 0))
+                                BOOST_VERIFY_MSG(numThreads == iBlockThreadCount,
```

Comment:
Could we keep the `throw` for now instead of the `BOOST_VERIFY_MSG`?
We should do such a change consistently across all places after incestigating all options.
E.g. the doc says `BOOST_VERIFY_MSG(expr,msg) expands to ((void)(expr)) when NDEBUG is defined, to BOOST_ASSERT_MSG(expr,msg) when it's not`. This would be a change in behaviour because the original version throws when the `ALPAKA_DEBUG` level is high enough in both, release and debug builds.

Reply:
Done. That would have completely circumvent my intention anyway, thanks. Just to get it right, do you mean, that the `BOOST_VERIFY` in `ExecCpuOmp2Blocks.hpp` should be always runtime/compile time error?

Reply:
A new PR or just a second commit into this one?

Reply:
2nd commit should be good.